Sources/JWTKit/JWTHeader.swift

```diff
@@ -1,20 +1,126 @@
 /// The header (details) used for signing and processing the JWT.
-package struct JWTHeader: Codable, Sendable {
+public struct JWTHeader: Codable, Sendable {
     /// The algorithm used with the signing.
-    package var alg: String?
+    public var alg: String?
 
     /// The Signature's Content Type.
-    package var typ: String?
+    public var typ: String?
 
     /// The Payload's Content Type.
-    var cty: String?
-
-    /// Critical fields.
-    var crit: [String]?
+    public var cty: String?
 
     /// The JWT key identifier.
-    var kid: JWKIdentifier?
+    public var kid: JWKIdentifier?
 
     /// The x5c certificate chain.
-    package var x5c: [String]?
+    public var x5c: [String]?
+
+    /// Custom fields.
+    public var customFields: [String: JWTHeaderField]?
+
+    init() {}
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encodeIfPresent(self.alg, forKey: .alg)
+        try container.encodeIfPresent(self.typ, forKey: .typ)
+        try container.encodeIfPresent(self.cty, forKey: .cty)
+        try container.encodeIfPresent(self.kid, forKey: .kid)
+        try container.encodeIfPresent(self.x5c, forKey: .x5c)
+
+        if let customFields = self.customFields {
+            try customFields.forEach { key, value in
+                try container.encode(value, forKey: .custom(name: key))
+            }
+        }
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.alg = try container.decodeIfPresent(String.self, forKey: .alg)
+        self.typ = try container.decodeIfPresent(String.self, forKey: .typ)
+        self.cty = try container.decodeIfPresent(String.self, forKey: .cty)
+        self.kid = try container.decodeIfPresent(JWKIdentifier.self, forKey: .kid)
+        self.x5c = try container.decodeIfPresent([String].self, forKey: .x5c)
+
+        let excludedKeys: Set<String> = ["alg", "typ", "cty", "kid", "x5c"]
```

Comment:
My only complaint is I don't like having these hardcoded here - can we make this a property on the `CodingKeys` enum instead?